models/ed/R/SAS.ED2.R

```diff
@@ -0,0 +1,502 @@
+  # ========================
+  # Moisture Limitation 
+  # ========================
+  # rel_soil_moist <- sum(swc_y*area.dist)
+  if(decomp_scheme %in% c(0,1)){
+    if (rel_soil_moist <= resp_opt_water){
+      water_limitation = exp( (rel_soil_moist - resp_opt_water) * resp_water_below_opt)
+    } else {
+      water_limitation = exp( (resp_opt_water - rel_soil_moist) * resp_water_above_opt)
+    }
+  } else if(decomp_scheme==2){
+    # Dry soil Limitation
+    lnexpdry <- rh_decay_dry * (rh_dry_smoist - rel_soil_moist)
+    lnexpdry <- max(-38, min(38, lnexpdry))
+    smdry_fun <- 1+exp(lnexpdry)
+    
+    # Wet Soil limitation
+    lnexpwet <- rh_decay_wet * (rel_soil_moist - rh_wet_smoist)
+    lnexpwet <- max(-38, min(38, lnexpwet))
+    smwet_fun <- 1+exp(lnexpwet)
+    
+    water_limitation <- 1/(smdry_fun * smwet_fun)
+  } else {
+    water_limitation = rel_soil_moist * 4.0893 - rel_soil_moist**2 * 3.1681 - 0.3195897
+  }
+  # ========================
+  
+  A_decomp <- temperature_limitation * water_limitation # aka het_resp_weight
+  # *************************************
+  
+  # *************************************
+  # Calculate the steady-state pools
+  # NOTE: Current implementation weights carbon input by patch size rather than using the 
+  #       carbon balance from the oldest state (as was the first implementation)
+  # *************************************
+  # -------------------
+  # Do the carbon and fast nitrogen pools
+  # -------------------
+  fsc_ss <- fsc_in_y[length(fsc_in_y)]/(fsc_loss * A_decomp)
+  ssl_ss <- ssl_in_y[length(ssl_in_y)]/(ssl_loss * A_decomp * Lc) # Structural soil C
+  ssc_ss <- ((ssl_loss * A_decomp * Lc * ssl_ss)*(1 - r_stsc))/(ssc_loss * A_decomp )
+  fsn_ss <- fsn_in_y[length(fsn_in_y)]/(fsc_loss * A_decomp)
+  # -------------------
+  
+  # -------------------
+  # Do the mineralized nitrogen calculation
+  # -------------------
+  #ED2: csite%mineralized_N_loss  = csite%total_plant_nitrogen_uptake(ipa)             
+  # + csite%today_Af_decomp(ipa) * Lc * K1 * csite%structural_soil_C(ipa)                     
+  # * ( (1.0 - r_stsc) / c2n_slow - 1.0 / c2n_structural)
+  msn_loss <- pln_up_y[length(pln_up_y)] + 
+    A_decomp*Lc*ssl_loss*ssl_in_y[length(ssl_in_y)]*
+    ((1.0-r_stsc)/c2n_slow - 1.0/c2n_structural)
+  
+  #fast_N_loss + slow_C_loss/c2n_slow
+  msn_med  <- fsc_loss*A_decomp*fsn_in_y[length(fsn_in_y)]+ (ssc_loss * A_decomp)/c2n_slow 
+  
+  msn_ss   <- msn_med/msn_loss
+  # -------------------
+  # *************************************
+  
+  # *************************************
+  # Replace dummy values in patch matrix with the steady state calculations
+  # *************************************
+  # Figure out what steady state value index we shoudl use 
+  # Note: In the current implementaiton this should be 1 because we did the weighted averaging up front, 
+  #       but if something went wrong and dimensions are off, use this to pick the last (etc)
+  p.use <- 1
```

Comment:
Not following this comment. To the extent that I can follow the code above, the variables below look like they should be scalars. And if they're not, I don't follow why we'd want to use the first value.